src-v5/carousel.tsx

```diff
@@ -49,12 +49,20 @@ export const Carousel = (props: CarouselProps): React.ReactElement => {
     );
     to && props.beforeSlide(slide, endSlide);
     !props.disableAnimation && setAnimation(true);
-
     setCurrentSlide(to ?? currentSlide);
     setTimeout(
       () => {
         if (!isMounted.current) return;
-        to && props.afterSlide(currentSlide);
+
+        let nextIndex = to ?? currentSlide;
+        if (nextIndex < 0) {
+          nextIndex += count;
+        }
+        if (nextIndex === count) {
+          nextIndex = 0;
+        }
+
+        typeof to === 'number' && props.afterSlide(nextIndex);
```

Comment:
previously just doing the check for `to` would actually fail it if the slide index is `0` since that is normalized to `false`.

Reply:
I think the same thing will happen for `beforeSlide`. Could we fix this as well?